### PR TITLE
feat: adapt for using jenkins as build system and integration for gitlab pipeline

### PR DESCRIPTION
**Pull request by Tristar666**
Check if MR cannot be merged for gitlab automerge
Fix bug with cache while getting information about MR for gitlab automerge.

## Changes:
Increase number of attempts to 100 for gitlab automerge
Check if MR cannot be merged for gitlab automerge
Fix bug with cache while getting information about MR for gitlab automerge.

## Context:

We use jenkins as build system with gitlab plugin for integration with gitlab pipelines. To start jenkins job we use web hooks and for a while build start and create pipeline in gitlab using jenkins plugin. In that case i increased number of max attempts for gitlab automerge. 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/platform/gitlab/index.ts [hunk 1]
```diff
@@ -432,18 +432,25 @@ async function tryPrAutomerge(
   if (platformOptions?.gitLabAutomerge) {
     try {
       const desiredStatus = 'can_be_merged';
-      const retryTimes = 5;
+      const cannotBeMerged = `cannot_be_merged`;
+      const retryTimes = 100;
```
Comment (viceice) [inline lib/platform/gitlab/index.ts:436]:
```suggestion
      const retryTimes = 5;
```

No more that 5 tries, increase delay instead.

**Review by viceice — CHANGES_REQUESTED** [review]
Needs tests

**Comment by viceice** [comment]
needs cla signing first

File: lib/platform/gitlab/index.ts (hunk 1 above)
Comment (Tristar666) [reply lib/platform/gitlab/index.ts:436]: 15 seconds is preferable for you?

Comment (viceice) [reply lib/platform/gitlab/index.ts:436]: I think we should not retry more that 5 times, but instead increase the delay computation.

**Comment by viceice** [comment]
You have some mismatch with your github account and the commiter email, please fix it, otherwise we can't merge this pr because of not complete cla signed

**Pull request closed without merging**